components/brave_ads/browser/notification_helper_android.cc

```diff
@@ -37,14 +37,13 @@ bool NotificationHelperAndroid::ShouldShowNotifications() {
   bool is_notifications_enabled = (status == kAppNotificationsStatusEnabled ||
       status == kAppNotificationStatusUndeterminable);
 
-  bool is_notification_channel_enabled = IsBraveAdsNotificationChannelEnabled();
-
-  auto is_foreground = BackgroundHelper::GetInstance()->IsForeground();
-  auto should_show_notifications =
-      CanShowBackgroundNotifications() || is_foreground;
-
-  return is_notifications_enabled && is_notification_channel_enabled &&
-      should_show_notifications;
+  bool is_foreground = BackgroundHelper::GetInstance()->IsForeground();
+  bool is_notification_channel_enabled = IsBraveAdsNotificationChannelEnabled(is_foreground);
+  bool result = is_notifications_enabled && is_notification_channel_enabled;
+  if (!is_foreground) {
+    result = result && CanShowBackgroundNotifications();
```

Comment:
See above comment as `CanShowBackgroundNotifications` business logic should also be moved to `AdsImpl::CheckReadyAdServe`

Reply:
I will raise a ticket to move this business logic as discussed